components/NavigationBar/src/MDCNavigationBar.m

```diff
@@ -317,8 +317,13 @@ - (void)layoutSubviews {
   CGRect alignedFrame = [self mdc_frameAlignedVertically:titleFrame
                                             withinBounds:textFrame
                                                alignment:titleVerticalAlignment];
-  _titleLabel.frame =
-      [self mdc_frameAlignedHorizontally:alignedFrame alignment:self.titleAlignment];
+  alignedFrame = [self mdc_frameAlignedHorizontally:alignedFrame alignment:self.titleAlignment];
+  CGFloat scale = self.window.screen.scale;
```

Comment:
Perhaps we want MDC*ToScale() to treat a scale of 0.0 as if it were equal to device.screen.scale?
We could avoid checking for 0.0 every time.
Might be an issue for people outputting to a second screen via airplay?

Reply:
Since this is a C-function, it's difficult to know [which of the screens](https://developer.apple.com/documentation/uikit/uiscreen/1617812-screens) we should use for the scale factor.  It would work as an extension on UIView, though.

Reply:
I think between returning an unaligned frame and trying to guess which scale to use, it might make sense to just default to 1x if it's not provided - at least then it'll point-align and be guaranteed to be pixel-aligned.

Reply:
I'm still not convinced this is really a problem and I don't want to clutter up the codebase with a check for it. We use scale all over the place.

Reply:
Yes, I think we're in agreement that the check should not be necessary for clients. I'd like to change the function to be safe (it currently returns CGRectNull), so  I hope we're discussing what default behavior to provide.

My recommendation is to return the rect aligned to a 1x scale, though we could just return the original rect instead.